osi_referenceline.proto

```diff
@@ -87,31 +111,57 @@ message ReferenceLine
     // ## Adding T coordinates
     //
     // To describe points that are not directly on the polyline, a T coordinate
-    // is added. T is the signed 2D distance (i.e. <code>hypot(A.X-B.X,
-    // A.Y-B.Y)</code>, if A and B are the two points) between the point to
-    // describe and the nearest point on the polyline (this point might either
-    // be on a line segment or at an edge between two line segments). The
-    // distance is positive if the point is left of the polyline (in definition
-    // direction), negative if it is right of it.
-    // The S position of such a point outside the reference line is the same as
-    // the S value of the nearest point on the polyline.
-    //
-    // Notes:
-    // - The "nearest point on the polyline" is determined in 3D (even if the
-    //   resulting T value is only the 2D distance), in order to choose the
-    //   correct point for 3D curves (think reference lines for roads in parking
-    //   decks).
-    // - If there are several "nearest points", the one with the smallest S
-    //   coordinate on the polyline is chosen.
+    // is added. T is the signed 2D distance between the point to describe (P)
+    // and a projected point (P_proj) on the polyline. There are two ways of
+    // defining this point, depending on the ReferenceLine type.
+    //
+    // ## Nearest point (TYPE_POLYLINE)
+    //
+    // The projection point is the nearest point on the polyline (this point might
+    // either be on a line segment or at an edge between two line segments).
+    //
+    // ## T axis definition (TYPE_POLYLINE_WITH_T_AXIS)
+    //
+    // The T axis (projecting axis) is the line going through P and the
+    // intersection point (I). I is defined as the intersection of both
+    // T axes of two consecutive ReferenceLinePoints (see example and
+    // image below for illustration). If both T axes of the neighboring
+    // ReferenceLinePoint are parallel (so no intersection point exists), the
+    // resulting T axis direction is equal to the T axis of these ReferenceLinePoints.
+    // The T coordinate of the point in question is then defined as
+    // <code>hypot(P.X-P_proj.X,P.Y-P_proj.Y)</code>. The projected point P_proj
+    // might either be on a line segment or at an edge between two line segments.
+    // The distance is positive if the point is left of the polyline (in
+    // definition direction), negative if it is right of it. The S position of
+    // such a point outside the reference line is the same as the S value of the
+    // projected point on the polyline, resulting in all points on a single T
+    // axis having the same S coordinate.
+    //
+    // \note The ST coordinate system shall be defined in 2D only. When
```

Comment:
Note that this explicitly is different for the old mode (and thus should be mentioned). There, the nearest point is determined in 3D, even if the distance is 2D. I think this makes much more sense, e.g. if you have a spiraling upward road in a multi-storey parking lot, you really do want to get the nearest point in 3D. Otherwise the S position will make no sense. So I think the question is whether you really want to define this only in 2D. See the previous definition on line 100.

Reply:
removed 2D restriction again